### PR TITLE
scripts:generic_variables.mk Fix NO_OS_VERSION

### PR DESCRIPTION
**Pull request by danmois**
Rewrite the NO_OS_VERSION compiler define and modify iio.c so that it is correctly included in the xml header.

Fixes: fdfaea1c8a6e

Signed-off-by: George Mois <george.mois@analog.com>

### CODE REVIEW

File: iio/iio.c [hunk 1]
```diff
@@ -73,6 +73,9 @@
 #define IIOD_CONN_BUFFER_SIZE	0x1000
 #define NO_TRIGGER				(uint32_t)-1
 
+#define STRINGIFY(x) #x
```
Comment (buha) [inline iio/iio.c:76]: elegant solution but please move these 2 macros in no_os_util.h, they will be useful everywhere

Comment (buha) [reply iio/iio.c:76]: and obviously prefix them

**Review by dbogdan — APPROVED** [review]

**Comment by danmois** [comment]
V2
- place defines in no_os_util

**Comment by buha** [comment]
cppcheck, astyle fail

**Comment by danmois** [comment]
V3

- run astyle

**Comment by danmois** [comment]
V4
- remove defines from no_os_util.h (cppcheck fail)

**Pull request merged**